src/Compilers/VisualBasic/Test/Semantic/Semantics/RequiredMembersTests.vb

```diff
@@ -0,0 +1,1283 @@
+            72 65 20 6e 6f 74 20 73 75 70 70 6f 72 74 65 64
+            20 69 6e 20 74 68 69 73 20 76 65 72 73 69 6f 6e
+            20 6f 66 20 79 6f 75 72 20 63 6f 6d 70 69 6c 65
+            72 2e 01 00 00
+        )
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.CompilerFeatureRequiredAttribute::.ctor(string) = (
+            01 00 0f 52 65 71 75 69 72 65 64 4d 65 6d 62 65
+            72 73 00 00
+        )
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance int32 get_Prop () cil managed 
+    {
+        ldarg.0
+        ldfld int32 S::f
+        ret
+    }
+
+    .method public hidebysig specialname 
+        instance void set_Prop (
+            int32 'value'
+        ) cil managed 
+    {
+        ldarg.0
+        ldarg.1
+        stfld int32 S::f
+        ret
+    }
+
+    .property instance int32 Prop()
+    {
+        .custom instance void [mscorlib]System.Runtime.CompilerServices.RequiredMemberAttribute::.ctor() = (
+            01 00 00 00
+        )
+        .get instance int32 S::get_Prop()
+        .set instance void S::set_Prop(int32)
+    }
+}
+"
+
+            Dim ilAssembly = CompileIL(il)
+
+            Dim comp = CreateCompilation("
+Module M
+    Sub Main()
+        Dim s = New S()
+    End Sub
+End Module
+", {ilAssembly}, targetFramework:=TargetFramework.Net70)
+
+            comp.AssertTheseDiagnostics(<expected>
+BC37321: Required member 'Public Overloads Property Prop As Integer' must be set in the object initializer or attribute arguments.
+        Dim s = New S()
+                    ~
+                                        </expected>)
+        End Sub
+
+        <Fact>
+        Public Sub RequiredMemberAttributeDisallowedInSource()
+            Dim comp = CreateCompilation("
+Imports System.Runtime.CompilerServices
+<RequiredMember>
+Public Class C
+    <RequiredMember>
+    Public Property P As Integer
+
+    <RequiredMember>
+    Public F As Integer
+End Class", targetFramework:=TargetFramework.Net70)
+
+            comp.AssertTheseDiagnostics(<expected><![CDATA[
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+<RequiredMember>
+ ~~~~~~~~~~~~~~
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+    <RequiredMember>
+     ~~~~~~~~~~~~~~
+BC37325: 'System.Runtime.CompilerServices.RequiredMemberAttribute' is reserved for compiler usage only.
+    <RequiredMember>
+     ~~~~~~~~~~~~~~]]></expected>)
+        End Sub
+
+        <Fact>
+        Public Sub TupleWithRequiredFields()
+            Dim csharpComp = CreateCSharpCompilation("
+namespace System
+{
+    public struct ValueTuple<T1, T2>
+    {
+        public required T1 Item1;
+        public required T2 Item2;
+        public required int AnotherField;
+        public required int Property { get; set; }
+
+        public ValueTuple(T1 item1, T2 item2)
```

Comment:
>public ValueTuple(T1 item1, T2 item2) [](http://example.com/codeflow?start=8&length=37)

I think we should also test scenario with SetsRequiredMembers

Reply:
Talked offline. This will be entirely blocked (in C# as well).